src/core/remote/codecs.ts

```diff
@@ -0,0 +1,59 @@
+import { ExtensionCodec } from '@msgpack/msgpack';
+import { TypedArrayConstructor } from '@/src/types';
+
+// The extension type base offset
+const EXT_BASE = 0x01;
+
+// Constructor index determines extension type code
+// should match the ordering in codecs.py
+const TypedArrays = [
+  Uint8Array,
+  Uint8ClampedArray,
+  Uint16Array,
+  Uint32Array,
+  Int8Array,
+  Int16Array,
+  Int32Array,
+  Float32Array,
+  Float64Array,
+];
+
+function createTypedArrayCodec(
+  TypedArray: TypedArrayConstructor,
+  typeOffset: number,
+  extTypeBase: number = 0
+) {
+  return {
+    type: extTypeBase + typeOffset,
+    encode: (object: unknown) => {
+      if (object instanceof TypedArray) {
+        return new Uint8Array(
+          object.buffer,
+          object.byteOffset,
+          object.byteLength
+        );
+      }
+      return null;
+    },
+    decode: (data: Uint8Array) => {
+      // force a copy to handle alignment issues.
+      // TODO try to align typed array data when encoding
```

Comment:
```suggestion
```
Tracked #275

Reply:
I may just remove this TODO and leave it as a note. This is a harder problem due to msgpack upstream not providing the necessary context for how much alignment padding to inject.